backend/main.py

```
@@ -200,104 +200,143 @@ async def apply_migrations():
+            print("Please ensure DATABASE_URL is set in your Render dashboard.")
+            print("Go to: Dashboard > Service > Environment > DATABASE_URL")
+            raise ValueError("DATABASE_URL environment variable must be properly configured")
         
-        # Test database connection
-        if db_pool is not None:
-            async with db_pool.acquire() as conn:
-                await conn.fetchval("SELECT 1")
-            print("✅ Database connection test successful")
-        else:
-            raise Exception("Database pool is not initialized.")
+        print(f"🔗 Attempting to connect to existing Render database...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)